src/data_store/generic_data_store.cpp

```diff
@@ -100,8 +127,7 @@ void generic_data_store::setup() {
     }
     m_reader->set_save_minibatch_entries(false);
   }
-
-  m_minibatch_indices = &(m_reader->get_minibatch_indices());
+  m_my_minibatch_indices = &(m_reader->get_minibatch_indices());
```

Comment:
generic_data_reader::set_save_minibatch_entries() may take advantage of generic_data_reader::get_num_iterations_per_epoch() to reserve generic_data_reader::m_my_minibatch_indices